### DISCUSSION TITLE
Setting TextView color

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [tui-cs/Terminal.Gui#2884] [Q&A]
I think I need some assistance in understanding how color schemes work. I have a secondary form to display in my app.
![image](https://github.com/gui-cs/Terminal.Gui/assets/12089920/698915e5-63e7-4582-844b-e91bf7f27230)

I've figured out how to change the form color so that it stands out from the main form.

```powershell
    $cs = [Terminal.Gui.ColorScheme]::New()
    $cs.Normal = [Terminal.Gui.Attribute]::New([Terminal.Gui.Color]::BrightBlue,[Terminal.Gui.Color]::Gray)
    $subWin = [Terminal.Gui.Window]@{
        Title = 'Add Category'
        X = [Terminal.Gui.Pos]::Center()
        Y = [Terminal.Gui.Pos]::Center()
        Width = 50
        Height = 10
        ColorScheme =$cs
    }
```

However, I can't figure out how to format the TextView in the pop-up form to have a white/gray background and black text. For that matter, I'd like to get the check box control to not be contrasted with the popup background. 
If I create the popup without defining its color scheme, the controls are rendered the way I want, but then the popup background color is not what I want.

I am having trouble connecting the conceptual dots.

**Comment by jdhitsolutions** [comment]
I'm still not sure I fully understand the elements of ColorScheme, but I fumbled my way to a solution.

![image](https://github.com/gui-cs/Terminal.Gui/assets/12089920/0af3c6cb-7004-46dc-8339-98463face9a9)

```powershell
    $cs = [Terminal.Gui.ColorScheme]::New()
    $cs.Normal = [Terminal.Gui.Attribute]::New([Terminal.Gui.Color]::BrightBlue,[Terminal.Gui.Color]::Gray)
    $cs.Focus = [Terminal.Gui.Attribute]::New([Terminal.Gui.Color]::White,[Terminal.Gui.Color]::DarkGray)

    $subWin = [Terminal.Gui.Window]@{
        Title = 'Add Category'
        X = [Terminal.Gui.Pos]::Center()
        Y = [Terminal.Gui.Pos]::Center()
        Width = 50
        Height = 10
        ColorScheme =$cs
    }
```

And I don't set a color scheme on the text view controls.

**Reply by BDisp** [reply]
See this [link](https://gui-cs.github.io/Terminal.Gui/api/Terminal.Gui/Terminal.Gui.ColorScheme.html?q=colorscheme) which explain a little what each attribute does. Normally a view inherits the `ColorScheme` from the `SuperView`, but you can initialize a personalized `ColorScheme` fot the `TextField` as well and thus it use different colors for `Normal, Focus, Disabled, HotFocus, HotNormal`.

**Comment by tig** [comment]
You need to set all attributes of the ColorScheme, not just `Normal`.

In v2 you can use `new ColorScheme(oldScheme)` to create a new scheme based on another.

Why are you not just using `Dialog` for this sub-form?

**Reply by jdhitsolutions** [reply]
I have yet to find a clear explanation of the different color scheme elements. Normal and disabled are probably clear. But I don't fully understand the hot variants.

As for using `Dialog` I guess I never considered it. I'm using it elsewhere to display a block of text. I never considered using it with controls. What would be the advantage of using `Dialog` in this situation?

**Reply by BDisp** [reply]
The hot variants distinguish between the attribute of the hotkey (normally prefixed with the `_` or `~` on the `StatusBar`) when the view isn't focused (`HotNormal`) or focused (`HotFocus`).

**Reply by jdhitsolutions** [reply]
Thank you for that explanation. I was thinking "hot" in terms of active. I didn't make the hotkey connection. This makes so much more sense to me now.

**Comment by tig** [comment]
> I have yet to find a clear explanation of the different color scheme elements. Normal and disabled are probably clear. But I don't fully understand the hot variants.
> 
> As for using `Dialog` I guess I never considered it. I'm using it elsewhere to display a block of text. I never considered using it with controls. What would be the advantage of using `Dialog` in this situation?

One reason is Dialog already has a distinct color scheme.

Other reasons:
- Users are used to the standard look & feel of Dialog. If you roll your own, you will have inconsistency. For example your OK and Cancel buttons are not aligned in a way consistent with Dialog. 
- It is tested and robust. You are duplicating functionality that's already tested.

**Reply by jdhitsolutions** [reply]
I definitely prefer the look and feel of `Dialog`.

**Reply by tznind** [reply]
Another advantage to using `Dialog` is that in future end users will be able to customize look and feel from config json files.  Sticking to the global themes will ensure this works smoothly.
